viewing_party/party.py

```diff
@@ -1,23 +1,176 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+
+    if not (title and genre and rating):
+        return None
+    else:
+        movie = { "title" : title,
+                "genre" : genre,
+                "rating" : rating
+                }
+        return movie
+    
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            break
+
+    return user_data    
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    rating_list = []
+    if not user_data["watched"]:
+        return 0.0
+    for movie in user_data["watched"]:
+        rating_list.append(movie["rating"])
+    average_rating = sum(rating_list) / len(rating_list)
+    return average_rating
+
+
+def get_most_watched_genre(user_data):
+    if not user_data["watched"]:
+        return None
+    
+    genre_list = []
+    for movie in user_data["watched"]:
+        genre_list.append(movie["genre"])
+    
+    most_watched_genre_frequency = 0
+    most_watched_genre = genre_list[0]
+    for genre in genre_list:
+        genre_frequency = genre_list.count(genre)
+        if genre_frequency > most_watched_genre_frequency:
+            most_watched_genre_frequency = genre_frequency
+            most_watched_genre = genre
+
+    return most_watched_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+
+    user_watched_titles =[]
+    for movie in user_data["watched"]:
+        user_watched_titles.append(movie["title"])
+    
+    friends_watched_titles = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["title"] not in friends_watched_titles:
+                friends_watched_titles.append(movie["title"])
+    
+    user_unique_titles = []
+    for title in user_watched_titles:
+        if title not in friends_watched_titles:
+            user_unique_titles.append(title)
+    
+    user_unique_movies = []
+    for title in user_unique_titles:
+        for movie in user_data["watched"]:
+            if movie["title"] == title:
+                user_unique_movies.append(movie)
+
+    return user_unique_movies

```

Comment:
Another way to approach this problem would be to use sets. You could create a set of a set of a friend's watched titles. 

```python
for movie in user_data["watched"]: 
    if movie["title"] not in friends_watched_set: 
        user_unique_movies.append(movie) return unique_list